server/core/region.go

```diff
@@ -608,7 +608,51 @@ func (r *RegionsInfo) AddRegion(region *RegionInfo) []*RegionInfo {
 	}
 
 	r.regions.Put(region)
+	r.addRegionToSubTree(region)
+	return overlaps
+}
+
+// ReplaceOrAddRegion adds or replaces RegionInfo to regionTree and regionMap, also update leaders and followers by region peers
+func (r *RegionsInfo) ReplaceOrAddRegion(region *RegionInfo, overlaps []*RegionInfo, oldRegionItem *regionItem) {
+	// Add to tree and regions.
+	if !isEqualRegion(region, oldRegionItem) {
+		for _, item := range overlaps {
+			r.RemoveRegion(r.GetRegion(item.GetID()))
+		}
+		if origin := r.regions.Get(region.GetID()); origin != nil {
+			r.RemoveRegion(origin)
+		}
+		r.tree.update(region)
+	} else {
+		for _, item := range overlaps {
+			if item.GetID() == region.GetID() {
```

Comment:
Why pass this? the other meta information may changed.

Reply:
Does it need to update the tree?

Reply:
Because the overlaps will contain the old value of the current region, here is not to delete the old region, and take a workaround to delete the regions in all overlaps that are not corresponding to the current region. In general, this overlaps only There will be the old value of the current region. At this time, this region is not deleted from the tree, but the method of updating the region in the regionItem is used: code such as: oldRegionItem.region = region

Reply:
so should we add a line:
```
r.tree.update(region)
```

Reply:
I think  the line of 'oldRegionItem.region = region'  has an updated effect. Using this method (oldRegionItem.region = region) instead of another one(r.tree.update(region)) is also a key point for performance improvement.

Reply:
got